server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java

```diff
@@ -80,39 +82,43 @@ public QueryDataSet executeWithoutValueFilter(QueryContext context)
       timeFilter = ((GlobalTimeExpression) expression).getFilter();
     }
 
-    List<AggregateResult> aggregateResultList = new ArrayList<>();
     //TODO use multi-thread
-    for (int i = 0; i < selectedSeries.size(); i++) {
-      AggregateResult aggregateResult = aggregateOneSeries(i, timeFilter, context);
-      aggregateResultList.add(aggregateResult);
+    Map<Path, List<Integer>> seriesMap = mergeSameSeries(selectedSeries);
+    AggregateResult[] aggregateResultList = new AggregateResult[selectedSeries.size()];
+    for (Map.Entry<Path, List<Integer>> entry : seriesMap.entrySet()) {
+      List<AggregateResult> aggregateResults = aggregateOneSeries(entry, timeFilter, context);
+      int index = 0;
+      for (int i : entry.getValue()) {
+        aggregateResultList[i] = aggregateResults.get(index);
+        index++;
+      }
     }
-    return constructDataSet(aggregateResultList);
-  }
 
+    return constructDataSet(Arrays.asList(aggregateResultList));
+  }
 
-  /**
-   * get aggregation result for one series
-   */
-  private AggregateResult aggregateOneSeries(int i, Filter timeFilter, QueryContext context)
+  private List<AggregateResult> aggregateOneSeries(Map.Entry<Path, List<Integer>> series,
```

Comment:
add javadoc and an example

given : 
paths: s1, s2, s3, s1
aggregations: count, sum, count, sum

seriesMap:  
s1 -> 0, 3
s2 -> 2
s3 -> 3

Reply:
Hi, thanks! I have intended put the example to `mergeSameSeries` method

Reply:
this function could be renamed to  groupAggregationsBySeries

Reply:
Done.